src/actions/MastodonAddVariablesAction.C

```diff
@@ -22,47 +22,47 @@
 
 template <>
 InputParameters
-validParams<MastodonAddVariableAction>()
+validParams<MastodonAddVariablesAction>()
 {
   InputParameters params = validParams<Action>();
   params.addClassDescription("Create displacement Variables, and velocity, acceleration, stress "
                              "and strain AuxVariables automatically.");
 
   params.addParam<bool>(
-      "add_variables", true, "false, if displacement Variables are created manually.");
+      "variables", true, "false, if displacement Variables are created manually.");
   params.addParam<bool>(
-      "add_auxvariables",
+      "auxvariables",
       true,
       "false, if velocity, acceleration, stress and strain AuxVariables are created manually.");
```

Comment:
I would like a separate boolean for the strain and stress auxvariables. Instead of having separate booleans for setting up for output auxvariables and auxkernels, won't it be easier to use the same boolean and set up both if the boolean is true? That will make sure the variable names are consistent and I don't see why someone would want to set up the auxvariables on their own while using the automatic option for creating the corresponding auxkernels.

Also if the user uses first order Lagrange shape functions for the stress/strain auxvariables and tries to use it with the automatically generated RankTwoAux auxkernel, then an error will be generated like 
``
*** ERROR ***
Nodal AuxKernel 'stress_xy' attempted to reference material property 'rank_two_tensor'
Consider using an elemental auxiliary variable for 'stress_xy'.
``

Reply:
Agreed, I will change the booleans for auxvariables and auxkernels. 

Which is more widely used: first order or second order? We need to choose a default and keep it consistent through out the action and may be place error checks accordingly. I think we also need to assume that a user would either use all the defaults in the action or create all of these parameters manually. If not the action can become complicated and I don't think it would be worth it. What do you think @sveerara and @aeslaughter?

Reply:
I feel that the user should not be given the option of creating variables/auxvariables on their own when they want the corresponding kernels/auxkernels to be created automatically. 

How about something like this:
- displacement variables and stress divergence kernels are set up automatically if model block is created. These are required for all static/dynamic analysis.
- dynamics = true sets up velocity & acceleration auxvariables, auxkernels and inertia kernels. All are required for a dynamic analysis so it might be better to put all of them under the same option.
- stress_strain_output = true would set up stress/strain auxvariables and corresponding auxkernels.

if any of these are off, the user has to create both the variable and corresponding kernel/auxkernel on their own. That way we don't need to worry about checking the variable names.

The way you set first/second order for nodal variables based on the order of element in the mesh is correct. The stress and strain auxvariables have to be elemental auxvariables that are of type constant monomial irrespective of the element type used for disp/vel/accel. I just was saying that letting the user create stress/strain auxvariables on their own while setting up the corresponding auxkernel using the action might lead to inconsistencies in not just the name of the variable but also the type of the variable.

Reply:
@sveerara: Ok, I like that too and it looks like we are on the same page. I will make those changes next week. Also I think it is worth having a call between @aeslaughter, you and me to talk in more detail about the Mastodon block. Since I am not a heavy Mastodon user I need suggestions like above.

Reply:
Thanks @cbolisetti. A call to discuss this sounds good.